src/parser/state_machine/syntax/comment.rs

```diff
@@ -42,14 +42,14 @@ define_state_group!(comment_states_group = {
     }
 
     comment_less_than_sign_state {
-        b'!' => ( --> comment_less_than_sign_bang_state )
-        b'>' => ()
+        b'!' => ( mark_comment_text_end; --> comment_less_than_sign_bang_state )
+        b'<' => ( mark_comment_text_end; )
         eof  => ( reconsume in comment_state )
         _    => ( reconsume in comment_state )
     }
 
     comment_less_than_sign_bang_state {
-        b'-' => ( --> comment_less_than_sign_bang_dash_state )
+        b'-' => ( mark_comment_text_end; --> comment_less_than_sign_bang_dash_state )
```

Comment:
actually this is an interesting point I didn't fully figure out. it seems that `mark_comment_text_end` ends the current comment text at the _previous_ character. added a comment, hopefully it makes sense